docs/src/description/organization.md

```diff
@@ -0,0 +1,77 @@
+| [public inputs](./declarations.md#public-inputs-public_inputs)                        | required    | not allowed       |
+| [periodic columns](./declarations.md#periodic-columns-periodic_columns)               | optional    | optional          |
+| [random values](./declarations.md#random-values-random_values)                        | optional    | not allowed       |
+| [boundary constraints](./constraints.md#boundary-constraints-boundary_constraints)    | required    | not allowed       |
+| [integrity constraints](./constraints.md#integrity-constraints-integrity_constraints) | required    | not allowed       |
+| [evaluators](./evaluators.md)                                                         | optional    | required          |
+
+Note that constants and evaluators are not really distinct sections but rather a set of declarations which can be done in-between any other sections.
+
+### Root module
+A root module defines an entrypoint into an AirScript project. It must start with a name declaration which consists of a `def` keyword followed by the name of the AIR project. For example:
+```
+def ExampleAir
+```
+where the name of the module must:
+- Be a string consisting of alpha-numeric characters and underscores.
+- Start with a letter.
+- End with a newline.
+
+Besides the name declaration, a root module must:
+
+- Describe the shape of the execution trace (done via the *trace columns* section).
+  - If the trace consists of more than one segment (e.g., main and auxiliary segments), describe random values available to the prover after each segment commitment (done via the *random values* section).
+- Describe the shape of the public inputs (done via the *public inputs* section).
+- Describe the boundary constraints placed against the execution trace (done via the *boundary constraints* section).
+- Describe the integrity constraints placed against the execution trace (done via the *integrity constraints* section).
+
+To aid with boundary and integrity constraint descriptions, a root module may also contain definitions of constants, evaluators, and periodic columns.
+
+### Supporting modules
+Supporting modules can be used to split integrity constraint descriptions across multiple files. A supporting module must start with a name declaration which consists of a `mod` keyword followed by the name of the module. For example:
+
+```
+mod example_module
+```
+where the name of the module must:
+- Be the same as the name of the file in which the supporting module is defined (e.g., the above module must be located in `example_module.air` file).
+- Be a string consisting of alpha-numeric characters and underscores.
+- Start with a letter.
+- End with a newline.
+
+Besides the name declaration, supporting modules must include at least one *evaluator*. Evaluators defined in a supporting module may be imported by a root or other supporting modules.
```

Comment:
This isn't true currently, and not necessary to implement anyway, so we probably want to remove this.